crates/rolldown_error/src/events/commonjs_variable_in_esm.rs

```diff
@@ -0,0 +1,66 @@
+use arcstr::ArcStr;
+
+use crate::{diagnostic::Diagnostic, types::diagnostic_options::DiagnosticOptions};
+
+use super::BuildEvent;
+
+/// Only record start offset for efficiency and simplicity
+#[derive(Debug)]
+pub enum CjsExportStartOffset {
+  Module(u32),
+  Exports(u32),
+}
+impl CjsExportStartOffset {
+  pub fn start(&self) -> u32 {
+    match self {
+      CjsExportStartOffset::Module(start) | CjsExportStartOffset::Exports(start) => *start,
+    }
+  }
+
+  pub fn end(&self) -> u32 {
```

Comment:
Seems brings more complexity than it should just for saving a `u32` size.

Reply:
fine, if you think this is complex.

Reply:
Yeah. I'm thinking if we accept this PR, there's no reason about not applying this design principle for all possible places. So if we gonna refactor all our code with this design principle, the complexity it brings and bytes it save seems not worth.